events/models.py

```diff
@@ -84,6 +92,13 @@ class EventPromotion(models.Model):
     """Assume generic external user id for event owner/host"""
     user_external_id = models.CharField(db_index=True, max_length=128)
 
+    def _requires_list_clear(self):
```

Comment:
ditto seems like maybe this should be in hub app rather than on the model. also it's a bit confusing when list was already clear, this will still say it requires clear. maybe check if list is already cleared or not?